packages/fal/src/fal-transcription-model.ts

```diff
@@ -0,0 +1,380 @@
+      'su',
+      'sv',
+      'sw',
+      'ta',
+      'te',
+      'tg',
+      'th',
+      'tk',
+      'tl',
+      'tr',
+      'tt',
+      'uk',
+      'ur',
+      'uz',
+      'vi',
+      'yi',
+      'yo',
+      'yue',
+      'zh',
+    ])
+    .nullish(),
+
+  /**
+   * Whether to diarize the audio file. Defaults to true.
+   */
+  diarize: z.boolean().nullish(),
+
+  /**
+   * Level of the chunks to return. Either segment or word. Default value: "segment"
+   */
+  chunkLevel: z.enum(['segment', 'word']).nullish(),
+
+  /**
+   * Version of the model to use. All of the models are the Whisper large variant. Default value: "3"
+   */
+  version: z.enum(['3']).nullish(),
+
+  /**
+   * Default value: 64
+   */
+  batchSize: z.number().nullish(),
+
+  /**
+   * Number of speakers in the audio file. Defaults to null. If not provided, the number of speakers will be automatically detected.
+   */
+  numSpeakers: z.number().nullable().nullish(),
+});
+
+export type FalTranscriptionCallOptions = z.infer<
+  typeof falProviderOptionsSchema
+>;
+
+interface FalTranscriptionModelConfig extends FalConfig {
+  _internal?: {
+    currentDate?: () => Date;
+  };
+}
+
+export class FalTranscriptionModel implements TranscriptionModelV1 {
+  readonly specificationVersion = 'v1';
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  constructor(
+    readonly modelId: FalTranscriptionModelId,
+    private readonly config: FalTranscriptionModelConfig,
+  ) {}
+
+  private async getArgs({
+    providerOptions,
+  }: Parameters<TranscriptionModelV1['doGenerate']>[0]) {
+    const warnings: TranscriptionModelV1CallWarning[] = [];
+
+    // Parse provider options
+    const falOptions = parseProviderOptions({
+      provider: 'fal',
+      providerOptions,
+      schema: falProviderOptionsSchema,
+    });
+
+    // Create form data with base fields
+    const body: Omit<FalTranscriptionAPITypes, 'audio_url'> = {
+      task: 'transcribe',
+      diarize: true,
+      chunk_level: 'word',
+    };
+
+    // Add provider-specific options
+    if (falOptions) {
+      body.language = falOptions.language;
+      body.version = falOptions.version ?? undefined;
+      body.batch_size = falOptions.batchSize ?? undefined;
+      body.num_speakers = falOptions.numSpeakers ?? undefined;
+
+      if (typeof falOptions.diarize === 'boolean') {
+        body.diarize = falOptions.diarize;
+      }
+
+      if (falOptions.chunkLevel) {
+        body.chunk_level = falOptions.chunkLevel;
+      }
+    }
+
+    return {
+      body,
+      warnings,
+    };
+  }
+
+  async doGenerate(
+    options: Parameters<TranscriptionModelV1['doGenerate']>[0],
+  ): Promise<Awaited<ReturnType<TranscriptionModelV1['doGenerate']>>> {
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const { body, warnings } = await this.getArgs(options);
+
+    const base64Audio =
+      typeof options.audio === 'string'
+        ? options.audio
```

Comment:
do we know here that options.audio is a base64 string and not a url?

Reply:
IIRC URLs are downloaded in the AI core `transcribe.ts` before being passed to the model, so it should be `string | DataContent` where we've documented that string should be base64 encoded audio. But don't quote me 😂

Reply:
we can rely on base64 here, that's what we require per the spec